src/Vipps.net/VippsApi.cs

```diff
@@ -0,0 +1,70 @@
+﻿using System.Net.Http;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Logging.Abstractions;
+using Vipps.net.Infrastructure;
+using Vipps.net.Services;
+
+namespace Vipps.net
+{
+    public interface IVippsApi
+    {
+        IVippsAccessTokenService AccessTokenService();
+        IVippsEpaymentService EpaymentService();
+        IVippsCheckoutService CheckoutService();
+    }
+
+    public class VippsApi : IVippsApi
+    {
+        private readonly VippsConfigurationOptions _vippsConfigurationOptions;
+        private VippsHttpClient _vippsHttpClient;
+        private ILoggerFactory _loggerFactory;
+        private readonly VippsAccessTokenService _accessTokenService;
+
+        public static IVippsApi Create(
+            VippsConfigurationOptions options,
+            ILoggerFactory loggerFactory = null
+        )
+        {
+            return new VippsApi(options, loggerFactory);
+        }
+
+        private VippsApi(
+            VippsConfigurationOptions configurationOptions,
+            ILoggerFactory loggerFactory = null
+        )
+        {
+            _loggerFactory = loggerFactory ?? NullLoggerFactory.Instance;
+            this._vippsConfigurationOptions = configurationOptions;
+            _vippsHttpClient = new VippsHttpClient(new HttpClient(), configurationOptions);
+
+            _accessTokenService = new VippsAccessTokenService(
+                configurationOptions,
+                new AccessTokenServiceClient(_vippsHttpClient, configurationOptions),
+                new AccessTokenCacheService()
+            );
+        }
+
+        public IVippsAccessTokenService AccessTokenService()
+        {
+            return this._accessTokenService;
+        }
+
+        public IVippsEpaymentService EpaymentService()
```

Comment:
Any reason for instantiating the services through functions? Can't they live as a member of the VippsApi class, instantiated by the constructor?

Reply:
It could be, but if you want to dependency inject IVipps API I think you need to expose the different services thorugh the functions anyways. You could still instantiate them in the constructor, but I thought this way makes the code more readable. Especially if we later want to add different loggers in the services like this: 
`   public IVippsCheckoutService CheckoutApi()
        {
            var logger = _loggerFactory.CreateLogger<CheckoutServiceClient>();
            return new CheckoutService(new CheckoutServiceClient(logger, _vippsHttpClient, configurationProvider));
        }
`

Reply:
Not anymore, I changed this now.  Also did some changes to startup in the Vipps.net.AspCoreDemo, to use the IOptions pattern instead of injecting singleton.